lib/analyse.mli

```diff
@@ -1,9 +1,12 @@
 module Analysis : sig
+  type ocamlformat_version = Version of string | Vendored
+  [@@deriving yojson, eq]
```

Comment:
This could be defined in `Analyse_ocamlformat` only and still be part of the API. Also, the derivings are necessary in the implementation and in the tests but the tests could redefine them if we want to remove them from the API.